src-test/com/zone5cloud/retrofit/core/TestOAuthAPI.java

```diff
@@ -243,4 +244,58 @@ public void testAdhocToken() {
 			assertNotNull("Token should havea  scope", token.getScope());
 		}
 	}
+
+	@Test
+	public void testZone5Server_HasClientIdAndKey(){
+		Response<User> rsp = userApi.me().blockingFirst();
+		String clientId = rsp.raw().request().header("Api-Key");
+		String clientSecret = rsp.raw().request().header("Api-Key-Secret");
+
+		System.out.println("Client Id :"+clientId + "Client Secret : "+clientSecret);
+		assertNotNull(clientId);
+		assertNotNull(clientSecret);
+	}
+
+	@Test
+	public void testNonZone5Server_HasNoClientIdAndKey(){
+		Response<User> rsp = userApi.me().blockingFirst();
+		String clientId = rsp.raw().request().header("Api-Key");
+		String clientSecret = rsp.raw().request().header("Api-Key-Secret");
+
+		System.out.println("Client Id :"+clientId + "Client Secret : "+clientSecret);
+		assertNull(clientId);
+		assertNull(clientSecret);
+	}
+
+	@Test
+	public void testZone5Server_HasClientIdAndKey_WhenTokenRefreshed(){
+		login();
+		if(isSpecialized()) {
+			Response<OAuthToken> response = authApi.adhocAccessToken("wahooride").blockingSingle();
+			OAuthToken token = response.body();
+			assertNotNull(token);
+			String clientId = response.raw().request().header("Api-Key");
+			String clientSecret = response.raw().request().header("Api-Key-Secret");
+
+			System.out.println("Client Id :" + clientId + "Client Secret : " + clientSecret);
+			assertNotNull(clientId);
+			assertNotNull(clientSecret);
+		}
```

Comment:
This method name is misleading. Token refresh is achieved with authApi.refreshAccessToken(...) which is not being called in this unit test.

FYI adhocAccessToken("wahooride") is requesting an adhoc token to provide a third party (in this case a wahoo headunit) so that the third party can upload a file to the TP server. The returned OAuthToken is not the logged in users token.

Reply:
ok will change to refresh